api/sys_plugins.go

```diff
@@ -46,22 +46,14 @@ func (c *Sys) ListPlugins(i *ListPluginsInput) (*ListPluginsResponse, error) {
 	ctx, cancelFunc := context.WithCancel(context.Background())
 	defer cancelFunc()
 	resp, err := c.c.RawRequestWithContext(ctx, req)
-	if err != nil {
+	if err != nil && resp == nil {
```

Comment:
If `resp == nil` we will panic below when we call `resp.StatusCode`. I think this logic needs reworked a bit, maybe we can check the StatusCode here (if `resp != nil`) and if it's a 405 continue, otherwise return the error

Reply:
From looking at the code of RawRequestWithContext I can say, that `resp` always set if there is no error returned.